worker/repository/tasks.py

```diff
@@ -9,23 +9,28 @@
 staging_dir = os.environ['STAGING_DIR']
 
 
-@celery_app.task(bind=True, name="retrieve_from_staging", base=BaseTask)
-def task_retrieve_from_staging(self, object_id, job_id):
-    if not os.path.exists(staging_dir):
-        os.makedirs(staging_dir)
-    if not os.path.exists(working_dir):
-        os.makedirs(working_dir)
-    staging_path = os.path.join(staging_dir, object_id)
-    work_path = self.get_work_path(job_id)
-    if os.path.exists(work_path):
-        shutil.rmtree(work_path)
-    shutil.copytree(staging_path, work_path)
-
-
-@celery_app.task(bind=True, name="publish_to_repository", base=BaseTask)
-def task_publish_to_repository(self, object_id, job_id):
-    work_path = self.get_work_path(job_id)
-    repository_path = os.path.join(repository_dir, object_id)
-    if os.path.exists(repository_path):
-        shutil.rmtree(repository_path)
-    shutil.copytree(work_path, repository_path)
+class RetrieveFromStagingTask(BaseTask):
+
+    name = "retrieve_from_staging"
+
+    def execute_task(self):
+        staging_path = os.path.join(staging_dir, self.object_id)
+        work_path = self.get_work_path(self.job_id)
+        if os.path.exists(work_path):
+            shutil.rmtree(work_path)
+        shutil.copytree(staging_path, work_path)
+
+
+RetrieveFromStagingTask = celery_app.register_task(RetrieveFromStagingTask())
```

Comment:
Wieso ist hier eine Zuweisung und kein einfacher Aufruf?

Reply:
Ich habe das so aus der Celery-Doku übernommen, vermute aber, dass dies sicherstellt, dass beim Import der Klasse in anderen Files dann auf die von Celery dekorierte Klasse zugegriffen wird.

Reply:
Okay, also dann überall so?